deeplinkdispatch-base/src/main/java/com/airbnb/deeplinkdispatch/base/MatchIndex.java

```diff
@@ -308,7 +354,10 @@ private int getNextElementStartPosition(int elementStartPos, int parentBoundaryP
    * @return The first elementStartPos that is not part of the parent element anymore.
    */
   private int getElementBoundaryPos(int elementStartPos) {
-    return elementStartPos + HEADER_LENGTH + getValueLength(elementStartPos)
+    return elementStartPos
```

Comment:
would be nice to dedupe and chain this like `return getChildrenPos(elementStartPos) + getChildrenLength(elementStartPos)`